src/theme/components/tagSet.ts

```diff
@@ -1,8 +1,20 @@
+import { defineStyleConfig } from "@chakra-ui/react";
+import { createMultiStyleConfigHelpers } from "@chakra-ui/styled-system";
+import { defineStyle, StyleFunctionProps } from "@chakra-ui/system";
+
 import { activeFocus } from "./global";
 
-const TagSetFilter = {
-  parts: ["clearAll"],
-  baseStyle: ({ isDismissible }: { isDismissible: boolean }) => ({
+// This function creates a set of function that helps us
+// create multipart component styles.
+const { defineMultiStyleConfig, definePartsStyle } =
+  createMultiStyleConfigHelpers(["clearAll"]);
+
+interface TagSetFilterProps extends StyleFunctionProps {
+  isDismissible: boolean;
+}
+
+const TagSetFilter = defineMultiStyleConfig({
+  baseStyle: definePartsStyle(({ isDismissible }: TagSetFilterProps) => ({
```

Comment:
For some reason, this works, but from what I understand, it's not what Chakra intended. `definePartsStyle` expects an object whose keys are the components parts (https://github.com/chakra-ui/chakra-ui/blob/5ec0be610b5a69afba01a9c22365155c1b519136/packages/core/styled-system/src/define-styles.ts#L92). In other words, there shouldn't be top level styles, only top level parts. To fix this, we can update the parts array above to contain a "base" part:

const { defineMultiStyleConfig, definePartsStyle } =
  createMultiStyleConfigHelpers(["base", "clearAll"]);

And nest these current top level styles in an object under a key of `base`.